app/controllers/api/users.php

```diff
@@ -52,6 +53,13 @@ function createUser(string $hash, mixed $hashOptions, string $userId, ?string $e
             ? ID::unique()
             : ID::custom($userId);
 
+        if ($project->getAttribute('auths', [])['disallowPersonalData'] ?? false) {
```

Comment:
Would it be too much to define a single password validator in our DI that uses the current project to know what kind of rules to apply? It can help make the controllers much simpler and easier to work with and we can just inject this object directly to the param (DI works for params!).

Reply:
Done 👌